great_expectations/data_context/types/base.py

```diff
@@ -2101,10 +2101,22 @@ def update(
                 batch_request = self.batch_request
                 batch_request = batch_request or {}
                 runtime_batch_request = runtime_kwargs.get("batch_request")
-                if isinstance(runtime_batch_request, BatchRequest):
-                    batch_request = nested_update(
-                        batch_request, runtime_batch_request.to_json_dict()
+                if (
+                    runtime_batch_request.get("runtime_parameters") is not None
+                    and runtime_batch_request["runtime_parameters"].get("batch_data")
+                    is not None
+                ):
+                    if (
+                        batch_request.get("runtime_parameters") is not None
+                        and batch_request["runtime_parameters"].get("batch_data")
+                        is not None
+                    ):
+                        batch_request["runtime_parameters"].pop("batch_data")
+                    batch_data = runtime_batch_request["runtime_parameters"].pop(
+                        "batch_data"
                     )
+                    batch_request = nested_update(batch_request, runtime_batch_request)
```

Comment:
Handle DataFrames for `nested_update` when overriding a previously configured `Checkpoint` `batch_request` at runtime.

Reply:
Just to confirm my understanding : The reason why we aren't storing the value in line #2114 is because we are intending to over-ride the existing configuration?

Reply:
That is correct. The runtime `batch_data` can't just be replaced using the `nested_update` because with either `batch_request` or `runtime_batch_request` containing a DataFrame in `batch_data` would result in a serialization error.